lib/service/modules/requestContext.js

```diff
@@ -0,0 +1,62 @@
+/** Copyright (c) 2016 Kinvey Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License'); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+
+module.exports = (metadata) => {
+  let string = null;
+  const numericParts = [Number.NaN, Number.NaN, Number.NaN];
+
+  if ((typeof metadata.clientAppVersion !== 'undefined' && metadata.clientAppVersion !== null)
```

Comment:
Since you're relying on `metadata.clientAppVersion` being a string (you're calling `split` on it below), you can shorten this to just `if (typeof metadata.clientAppVersion === 'string' && metadata.clientAppVersion.length > 0)`